### ISSUE TITLE
Interface keys can falsely violate `no-use-before-define`

### ISSUE THREAD

**Issue by ncknuna** [eslint/typescript-eslint-parser#435]
**What version of TypeScript are you using?**
2.6.2

**What version of `typescript-eslint-parser` are you using?**
11.0.0

**What code were you trying to parse?**

```ts
interface Foo {
    bar: string
}

const bar = 'blah'
```

**What did you expect to happen?**
The code to be considered valid from the perspective of `no-use-before-define`, or the README to call out that this is an unsupported case.

**What happened?**
The key `bar` in the interface on line 2 was flagged as being used before defined. If I removed the later `const bar...` assignment, the lint error went away.

**Comment by JamesHenry** [comment]
Thanks for reporting @ncknuna!
